File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/CS/Direct2D_3D/SciFiTextDemo/D3DStructs.cs
// Copyright (c) Microsoft Corporation.  All rights reserved.

using System;
using System.Runtime.InteropServices;
using Microsoft.WindowsAPICodePack.DirectX.Direct3D;

namespace SciFiTextDemo
{
    [StructLayout(LayoutKind.Sequential)]
    public struct SimpleVertex
    {
        [MarshalAs(UnmanagedType.Struct)]
        public Vector3F Pos;
        [MarshalAs(UnmanagedType.Struct)]
        public Vector2F Tex;
    }


    public class VertexData
    {
        public Vertices VerticesInstance = new Vertices();
        public Indices IndicesInstance = new Indices();

        [StructLayout(LayoutKind.Sequential)]
        public class Vertices
        {
            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
            private SimpleVertex[] vertices = 
            {
               new SimpleVertex() { Pos = new Vector3F ( -1.0f, -1.0f, 16.0f ), Tex = new Vector2F( 0.0f, 0.0f )}, // upper-left
               new SimpleVertex() { Pos = new Vector3F (  1.0f, -1.0f, 16.0f ), Tex = new Vector2F( 1.0f, 0.0f )}, // upper-right
               new SimpleVertex() { Pos = new Vector3F (  1.0f, -1.0f, 0.0f ), Tex = new Vector2F( 1.0f, 1.0f )}, // lower-right
               new SimpleVertex() { Pos = new Vector3F ( -1.0f, -1.0f, 0.0f ), Tex = new Vector2F( 0.0f, 1.0f )} // lower-left
            };
        }

        [StructLayout(LayoutKind.Sequential)]
        public class Indices
        {
            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 6)]
            private ushort[] indices = 
            {
                3,1,0,
                2,1,3,
            };
        }
    }
}

File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/VB/Direct2D_3D/SciFiTextDemo/D3DStructs.vb
' Copyright (c) Microsoft Corporation.  All rights reserved.


Imports Microsoft.VisualBasic
Imports System
Imports System.Runtime.InteropServices
Imports Microsoft.WindowsAPICodePack.DirectX.Direct3D

Namespace SciFiTextDemo
	<StructLayout(LayoutKind.Sequential)> _
	Public Structure SimpleVertex
		<MarshalAs(UnmanagedType.Struct)> _
		Public Pos As Vector3F
		<MarshalAs(UnmanagedType.Struct)> _
		Public Tex As Vector2F
	End Structure


	Public Class VertexData
		Public VerticesInstance As New Vertices()
		Public IndicesInstance As New Indices()

		<StructLayout(LayoutKind.Sequential)> _
		Public Class Vertices
			<MarshalAs(UnmanagedType.ByValArray, SizeConst := 4)> _
			Private vertices() As SimpleVertex = { New SimpleVertex() With {.Pos = New Vector3F (-1.0f, -1.0f, 16.0f), .Tex = New Vector2F(0.0f, 0.0f)}, New SimpleVertex() With {.Pos = New Vector3F (1.0f, -1.0f, 16.0f), .Tex = New Vector2F(1.0f, 0.0f)}, New SimpleVertex() With {.Pos = New Vector3F (1.0f, -1.0f, 0.0f), .Tex = New Vector2F(1.0f, 1.0f)}, New SimpleVertex() With {.Pos = New Vector3F (-1.0f, -1.0f, 0.0f), .Tex = New Vector2F(0.0f, 1.0f)} }
		End Class

		<StructLayout(LayoutKind.Sequential)> _
		Public Class Indices
			<MarshalAs(UnmanagedType.ByValArray, SizeConst := 6)> _
			Private indices() As UShort = { 3,1,0, 2,1,3 }
		End Class
	End Class
End Namespace
